sdk/appconfiguration/Azure.ApplicationModel.Configuration/src/ConfigurationClient.cs

```diff
@@ -113,8 +123,13 @@ public virtual async Task<Response<ConfigurationSetting>> AddAsync(Configuration
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         public virtual Response<ConfigurationSetting> Add(ConfigurationSetting setting, CancellationToken cancellationToken = default)
         {
-            using (Request request = CreateAddRequest(setting))
+            using ClientDiagnosticScope scope = _pipeline.Diagnostics.CreateScope("ConfigurationClient.Add");
+            scope.AddAttribute("key", setting.Key);
```

Comment:
The `Create*Request` methods were adding null checks for `setting` that will be turned into null refs here.